vassal-app/src/main/java/VASSAL/build/module/properties/ChangePropertyButton.java

```diff
@@ -111,29 +112,35 @@ public PropertyChanger getPropertyChanger() {
 
   @Override
   public String[] getAttributeDescriptions() {
-    return ArrayUtils.addAll(
-      super.getAttributeDescriptions(),
-      Resources.getString("Editor.report_format"),
-      Resources.getString("Editor.ChangePropertyButton.options")
-    );
+    String[] descs = { Resources.getString("Editor.description_label") };
+    for (final String s : super.getAttributeDescriptions()) {
+      descs = ArrayUtils.add(descs, s);
+    }
+    descs = ArrayUtils.add(descs, Resources.getString("Editor.report_format"));
+    descs = ArrayUtils.add(descs, Resources.getString("Editor.ChangePropertyButton.options"));
+    return descs;
   }
```

Comment:
A few things here:

* Rather than a loop, you can use `ArrayUtils.addAll()` to do it in one shot: 

    ```descs = super.getAttributeDescriptions(descs, super.getAttributeDescriptions());```

That will be more efficient, because `ArrayUtils.add()` makes a new array on each call.

* But, this is still kind of crappy, making three new arrays when you could do it with one... I'm going to write a little utility function for this...

Reply:
I think this would do the same but with less generation of garbage?
```
final Stream.Builder<String> b = Stream.builder().add(Resources.getString("Editor.description_label"));
Arrays.stream(super.getAttributeDescriptions()).forEach(s -> b.add(s));
b.add(Resources.getString("Editor.report_format"))
 .add(Resources.getString("Editor.report_format"));

return b.build().toArray(String[]::new);
```